crates/sshx-server/src/web/socket.rs

```diff
@@ -95,15 +96,45 @@ async fn handle_socket(socket: &mut WebSocket, session: Arc<Session>) -> Result<
     session.sync_now();
     send(socket, WsServer::Hello(user_id, metadata.name.clone())).await?;
 
-    match recv(socket).await? {
-        Some(WsClient::Authenticate(bytes)) if bytes == metadata.encrypted_zeros => {}
+    let (user_guard, _) = match recv(socket).await? {
+        Some(WsClient::Authenticate(bytes, write_password_bytes)) => {
+            // `ct_eq` returns a `Choice`, and `unwrap_u8()` converts it to 1 (equal) or 0
+            // (not equal).
+            if bytes.ct_eq(metadata.encrypted_zeros.as_ref()).unwrap_u8() != 1 {
```

Comment:
<img width="1040" alt="image" src="https://github.com/user-attachments/assets/7d2b9de4-b98d-4046-a8a7-9695c45b434b" />

nit: shouldn't this do `into()` to use the From impl? I usually wouldn't bring it up except it seems like they bolded it for emphasis in the docs :)